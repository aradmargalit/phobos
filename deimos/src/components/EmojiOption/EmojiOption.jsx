import React from 'react';
import PropTypes from 'prop-types';
import { Select } from 'antd';

const { Option } = Select;

// This _should_ be a HOC, but Antd doesn't allow it, so here we are
export default function EmojiOption({ emoji, value, title }) {
  return (
    <Option value={value}>
      {title}
      <span className="emoji" role="img" aria-label={`${value} emoji`}>
        {emoji}
      </span>
    </Option>
  );
}

EmojiOption.propTypes = {
  emoji: PropTypes.string.isRequired,
  value: PropTypes.string.isRequired,
  title: PropTypes.string.isRequired,
};
