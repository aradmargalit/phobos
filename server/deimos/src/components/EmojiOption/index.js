import EmojiOption from './EmojiOption';

export default EmojiOption;
