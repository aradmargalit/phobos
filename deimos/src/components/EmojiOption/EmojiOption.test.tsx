import { Select } from 'antd';
import { shallow } from 'enzyme';
import React from 'react';

import EmojiOption from './EmojiOption';

const { Option } = Select;

const generateComponent = (props?) => {
  const defaultProps = {
    value: 'run',
    title: 'run',
  };

  const mergedProps = { ...defaultProps, ...props };

  return shallow(<EmojiOption {...mergedProps} />);
};

describe('<EmojiOption />', () => {
  it('renders without crashing', () => {
    const component = generateComponent();
    expect(component.find(Option)).toHaveLength(1);
    expect(component.find('span')).toHaveLength(1);
  });

  it('renders when no emoji matches', () => {
    generateComponent({
      title: 'handball',
      value: 'handball',
    });
  });
});
