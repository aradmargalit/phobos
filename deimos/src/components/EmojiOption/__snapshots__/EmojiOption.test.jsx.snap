// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<EmojiOption /> renders without crashing 1`] = `ShallowWrapper {}`;
